bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ModalImpl.java

```diff
@@ -0,0 +1,137 @@
+package com.adobe.cq.wcm.core.components.internal.models.v1;
+
+import javax.annotation.PostConstruct;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.jackrabbit.vault.util.JcrConstants;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.ModifiableValueMap;
+import org.apache.sling.api.resource.PersistenceException;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.models.annotations.Exporter;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.InjectionStrategy;
+import org.apache.sling.models.annotations.injectorspecific.SlingObject;
+import org.apache.sling.models.annotations.injectorspecific.ValueMapValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.adobe.cq.export.json.ComponentExporter;
+import com.adobe.cq.export.json.ExporterConstants;
+import com.adobe.cq.wcm.core.components.models.Modal;
+
+@Model(adaptables = SlingHttpServletRequest.class, adapters = { Modal.class,
+		ComponentExporter.class }, resourceType = ModalImpl.RESOURCE_TYPE)
+@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME, extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+public class ModalImpl implements Modal {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(ModalImpl.class);
+	protected static final String RESOURCE_TYPE = "core/wcm/components/modal/v1/modal";
+
+	private static final String KEY_MODAL_ID = "modalId";
+	private static final String HTML_EXT = ".html";
+
+	@SlingObject
+	private Resource resource;
+
+	@SlingObject
+	private ResourceResolver resourceResolver;
+
+	private String modalId;
+
+	@ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+	private String description;
+
+	@ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+	private boolean showModalByDefault;
+
+	@ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+	private String fragmentType;
+
+	@ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+	private String contentFragmentPath;
+
+	@ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+	private String experienceFragmentPath;
+
+	@PostConstruct
+	private void initModel() {
+		if (modalId == null) {
+			populateModalId();
+		}
+
+		if (StringUtils.isNotBlank(experienceFragmentPath) && !experienceFragmentPath.contains(HTML_EXT)) {
+			experienceFragmentPath = experienceFragmentPath.concat(HTML_EXT);
+		}
+	}
+
+	void populateModalId() {
+		String absoluteComponentPath = resource.getPath();
+		int index = absoluteComponentPath.indexOf(JcrConstants.JCR_CONTENT);
+		String relativeComponentPath = absoluteComponentPath.substring(index);
+		modalId = String.valueOf(Math.abs(relativeComponentPath.hashCode() - 1));
+
+		ModifiableValueMap map = resource.adaptTo(ModifiableValueMap.class);

```

Comment:
I wouldn't persist the modal ID in the repository.

Reply:
Modal Id is required to be shown in dialog. That's why we need to persist it.